keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt

```diff
@@ -112,27 +118,99 @@ class ApplicationLoadBalancerHandler(
         .map { it.await() }
     }
 
+  override suspend fun export(exportable: Exportable): SubmittedResource<ApplicationLoadBalancerSpec> {
+    val albs = cloudDriverService.getApplicationLoadBalancer(
+      account = exportable.account,
+      name = exportable.moniker.name,
+      regions = exportable.regions,
+      serviceAccount = exportable.serviceAccount
+    )
+
+    val zonesByRegion = albs.map { (region, alb) ->
+      region to cloudDriverCache.availabilityZonesBy(
+        account = exportable.account,
+        vpcId = cloudDriverCache.subnetBy(exportable.account, region, alb.location.subnet).vpcId,
+        region = region
+      )
+    }
+      .toMap()
+
+    val base = albs.values.first()
+    val spec = ApplicationLoadBalancerSpec(
+      moniker = base.moniker,
+      locations = SubnetAwareLocations(
+        account = exportable.account,
+        vpc = base.location.vpc,
+        subnet = base.location.subnet,
+        regions = albs.map { (region, alb) ->
+          SubnetAwareRegionSpec(
+            name = region,
+            availabilityZones =
+            if (alb.location.availabilityZones.containsAll(
```

Comment:
Do you think it's worth pulling out this calculation to before we're creating the `spec`? I think I like resolving most values first (especially if there's more complicated logic than like `substring()`.